apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -715,7 +808,20 @@ export const resourceRouter = router({
   searchWithResourceIds: protectedProcedure
     .input(searchWithResourceIdsSchema)
     .output(searchWithResourceIdsOutputSchema)
-    .query(async ({ input: { siteId, resourceIds } }) => {
+    .query(async ({ ctx, input: { siteId, resourceIds } }) => {
+      // Validate permissions concurrently for all resourceIds
+      // Note: might want to refactor this if performance becomes an issue
+      await Promise.all(
+        resourceIds.map((resourceId) =>
+          validateUserPermissionsForResource({
+            userId: ctx.user.id,
+            siteId: Number(siteId),
+            resourceId: resourceId ? String(resourceId) : null,
+            action: "read",
+          }),
+        ),
+      )
+
```

Comment:
[question] same as above, might want to define another method that does a single query rather than do this if it's easy

Reply:
update to use bulk validation where only 1 DB query to fetch all resources, but loop the validation on the application-layer (in https://github.com/opengovsg/isomer/pull/1164/commits/506b434a11fdd3e8f1842fd78bf1673c6fd8d9be). Added unit tests here https://github.com/opengovsg/isomer/pull/1164/commits/9d42baad0a7509448393766380defc503dbc11eb